src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -99,28 +100,29 @@ Transaction::pointer Transaction::transactionFromSQL (
     return tr;
 }
 
-Transaction::pointer Transaction::transactionFromSQLValidated(
-    boost::optional<std::uint64_t> const& ledgerSeq,
-    boost::optional<std::string> const& status,
-    Blob const& rawTxn,
-    Application& app)
+Transaction::pointer Transaction::load (uint256 const& id, Application& app, error_code_i& ec)
 {
-    auto ret = transactionFromSQL(ledgerSeq, status, rawTxn, app);
+    using op = boost::optional<ClosedInterval<uint32_t>>;
 
-    if (checkValidity(app.getHashRouter(),
-            *ret->getSTransaction(), app.
-                getLedgerMaster().getValidatedRules(),
-                    app.config()).first !=
-                        Validity::Valid)
-        return {};
+    return boost::get<pointer> (load (id, app, op {}, ec));
```

Comment:
No need for the type alias and `op{}`. Use `boost::none` here instead.

Reply:
Done.